ui/store/actions.ts

```diff
@@ -5085,6 +5086,39 @@ export function setName(
   }) as any;
 }
 
+/**
+ * To create a data deletion regulation for MetaMetrics data deletion
+ */
+export function createMetaMetricsDataDeletionTask(): ThunkAction<
+  void,
+  MetaMaskReduxState,
+  unknown,
+  AnyAction
+> {
+  return async (dispatch: MetaMaskReduxDispatch) => {
+    try {
+      await submitRequestToBackground('createMetaMetricsDataDeletionTask');
+      await forceUpdateMetamaskState(dispatch);
+    } catch (error) {
+      logErrorWithMessage(error);
+    }
+  };
+}
+
+/**
+ * To check the status of the current delete regulation.
+ */
+export async function checkDataDeletionTaskStatus(): Promise<DataDeletionResponse> {
```

Comment:
Hmm, the return type seems wrong here. There is no `DataDeletionResponse` returned

Reply:
updated